src/Scan.js

```diff
@@ -126,6 +126,16 @@ class Scan extends React.Component {
     }
   }
 
+  setBarcodeRef = el => {
+    this.barcodeRef = el;
+  }
+
+  setFocusInput = () => this.barcodeRef.current.focus();
```

Comment:
I would like to see function with check if this.barcodeRef is not null and call focus like in CheckIn.js instead of single line function